File: samples/sample-app-storage/src/main/java/com/azure/android/storage/sample/core/util/tokenrequest/TokenResponseCallback.java
package com.azure.android.storage.sample.core.util.tokenrequest;

import com.azure.android.storage.sample.core.credential.AccessToken;

/**
 * The callback that {@link TokenRequestObserver} uses to notify the result of token retrieval.
 */
public interface TokenResponseCallback {
    /**
     * Notify successful retrieval of access-token.
     *
     * @param token the retrieved token
     */
    void onToken(AccessToken token);

    /**
     * Notify the error occurred during access-token retrieval.
     *
     * @param t the error
     */
    void onError(Throwable t);
}

File: samples/sample-app-storage-kotlin/src/main/java/com/azure/android/storage/sample/kotlin/core/util/tokenrequest/TokenResponseCallback.kt
package com.azure.android.storage.sample.kotlin.core.util.tokenrequest

import com.azure.android.storage.sample.kotlin.core.credential.AccessToken

/**
 * The callback that [TokenRequestObserver] uses to notify the result of token retrieval.
 */
interface TokenResponseCallback {
    /**
     * Notify successful retrieval of access-token.
     *
     * @param token the retrieved token
     */
    fun onToken(token: AccessToken)

    /**
     * Notify the error occurred during access-token retrieval.
     *
     * @param t the error
     */
    fun onError(t: Throwable?)
}
